CedarJavaFFI/src/interface.rs

```diff
@@ -125,6 +129,43 @@ pub(crate) fn call_cedar(call: &str, input: &str) -> String {
     })
 }
 
+#[derive(Serialize, Deserialize)]
+struct ValidateEntityCall {
+    schema: Value,
+    entities: Value,
+}
+
+pub fn json_validate_entities(input: &str) -> serde_json::Result<String> {
+    let ans = validate_entities(input)?;
+    serde_json::to_string(&ans)
+}
+
+/// public string-based JSON interface to be invoked by FFIs. Takes in a `ValidateEntityCall` and (if successful)
+/// returns unit value () which is null value when serialized to json.
+pub fn validate_entities(input: &str) -> serde_json::Result<Answer> {
+    let validate_entity_call = from_str::<ValidateEntityCall>(&input)?;
+    match Schema::from_json_value(validate_entity_call.schema) {
+        Err(e) => Ok(Answer::fail_bad_request(vec![e.to_string()])),
+        Ok(schema) => {
+            match Entities::from_json_value(validate_entity_call.entities, Some(&schema)) {
+                Err(error) => {
+                    let err_message = match error {
+                        EntitiesError::Serialization(err) => err.to_string(),
+                        EntitiesError::Deserialization(err) => err.to_string(),
+                        EntitiesError::Duplicate(err) => err.to_string(),
+                        EntitiesError::TransitiveClosureError(err) => err.to_string(),
+                        EntitiesError::InvalidEntity(err) => err.to_string(),
+                    };
```

Comment:
Does `error.to_string()` not work?

Reply:
It  gives a generic message rather than detailed one.

Reply:
Ah, this is a case where the useful information is hidden inside the `help` text... let me find the right incantation to use to get things to print properly

Reply:
Understood, I will try it.

Reply:
I tried this and it seems that help msg is not provided by all of them. help return option and it is not set for some of them.